hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -52,7 +52,29 @@ def unlink(self):
     @api.multi
     def _check_state_on_write(self, values):
         """ Hook for extensions """
-        self._check_state()
+        if not self._timesheet_validate_write(values):
+            self._check_state()
+
+    @api.model
+    def _timesheet_validate_write(self, values):
+        """ Hook for extensions """
+        return all([field_name not in values for field_name in
+                    self._timesheet_protected_fields()])
+
+    @api.model
+    def _timesheet_protected_fields(self):
```

Comment:
I prefer to prefix with a verb for indicating what is this: `_get_timesheet_protected_fields`